src/cpp/benders/benders_mpi/include/BendersMPI.h

```diff
@@ -87,8 +87,14 @@ class BendersMpi : public BendersBase {
   void AllReduce(const T &in_value, T &out_value, Op op) const {
     mpi::all_reduce(_world, in_value, out_value, op);
   }
-  void BuildGatheredCuts(const SubProblemDataMap &subproblem_data_map,
-                         const Timer &walltime);
   virtual void GatherCuts(const SubProblemDataMap &subproblem_data_map,
                           const Timer &walltime);
+  void BroadCastVariablesIndices();
+  virtual void ComputeSubproblemsContributionToOuterLoopCriterion(
+      const SubProblemDataMap &subproblem_data_map);
+  void SolveSubproblem(
+      SubProblemDataMap &subproblem_data_map,
+      PlainData::SubProblemData &subproblem_data, const std::string &name,
+      const std::shared_ptr<SubproblemWorker> &worker) override;
+  void UpdateOuterLoopMaxCriterionArea();
```

Comment:
I have the feeling the the outer loop is now mixed with Benders, and the fact that the outer loop is a loop around Benders becomes unclear when reading the code

Reply:
The thing is that the "criterion" can refer to something that is not specific to the outer loop, so maybe the notion of criterion is not attached to the outer loop and refers to something more general

Reply:
what do you think on renaming:
UpdateOuterLoopMaxCriterionArea -> UpdateMaxCriterionArea